File: TransCoder_tokenized_test_set_functions/java/FIND_TRIPLETS_ARRAY_WHOSE_SUM_EQUAL_ZERO_1.java
static void findTriplets ( int arr [ ], int n ) {
  boolean found = false;
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    HashSet < Integer > s = new HashSet < Integer > ( );
    for ( int j = i + 1;
    j < n;
    j ++ ) {
      int x = - ( arr [ i ] + arr [ j ] );
      if ( s . contains ( x ) ) {
        System . out . printf ( "%d %d %d\n", x, arr [ i ], arr [ j ] );
        found = true;
      }
      else {
        s . add ( arr [ j ] );
      }
    }
  }
  if ( found == false ) {
    System . out . printf ( " No Triplet Found\n" );
  }
}

File: TransCoder_tokenized_test_set_functions/cpp/FIND_TRIPLETS_ARRAY_WHOSE_SUM_EQUAL_ZERO_1.cpp
void findTriplets ( int arr [ ], int n ) {
  bool found = false;
  for ( int i = 0;
  i < n - 1;
  i ++ ) {
    unordered_set < int > s;
    for ( int j = i + 1;
    j < n;
    j ++ ) {
      int x = - ( arr [ i ] + arr [ j ] );
      if ( s . find ( x ) != s . end ( ) ) {
        printf ( "%d %d %d\n", x, arr [ i ], arr [ j ] );
        found = true;
      }
      else s . insert ( arr [ j ] );
    }
  }
  if ( found == false ) cout << " No Triplet Found" << endl;
}